src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -211,10 +211,12 @@ private ConstantValue DefaultSyntaxValue
                     var completedOnThisThread = state.NotePartComplete(CompletionPart.EndDefaultSyntaxValue);
                     Debug.Assert(completedOnThisThread);
 
-                    if (binder is not null && parameterEqualsValue is not null && !_lazyDefaultSyntaxValue.IsBad)
+                    if (binder is not null &&
+                        !_lazyDefaultSyntaxValue.IsBad &&
```

Comment:
> !_lazyDefaultSyntaxValue.IsBad && [](start = 23, length = 34)

It feels like this condition is not appropriate for the purpose of ```NullableWalker.AnalyzeIfNeeded```. Why does it matter what the constant value is? It doesn't look like SemanticModel cares, which is likely to lead to an inconsistency.